### PR TITLE
Activity restructure

### PR DESCRIPTION
**Pull request by vanderpm**
Added butterKnife to simplify code
Added String Utils classes
reformated logic of auton and teleop activities

### CODE REVIEW

File: app/src/main/java/com/example/vande/scouting2017/TeleopActivity.java [hunk 1]
```diff
@@ -29,12 +31,16 @@
...
 import butterknife.ButterKnife;
+import utils.FormatStringUtils;
+import utils.StringUtils;
+import utils.ViewUtils;
 
 
-public class TeleopActivity extends AppCompatActivity {
+public class TeleopActivity extends AppCompatActivity implements View.OnKeyListener {
 
     public String heading, auton, teleop, message;
```
Comment (KyletheWolff) [inline app/src/main/java/com/example/vande/scouting2017/TeleopActivity.java]: Only auton is being used remove the rest. 

Also put the String variables under the @BindViews

File: app/src/main/java/com/example/vande/scouting2017/TeleopActivity.java [hunk 2]
```diff
@@ -89,79 +95,130 @@
...
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_teleop);
 
-
         ButterKnife.bind(this);
-        //Save Data button
 
         Bundle bundle = getIntent().getExtras();
-        auton = bundle.getString("auton");
+        auton = bundle.getString("auton_extra");
```
Comment (KyletheWolff) [inline app/src/main/java/com/example/vande/scouting2017/TeleopActivity.java]: use the public static variable in AutonActivity AUTON_STRING_EXTRA

File: app/src/main/java/com/example/vande/scouting2017/TeleopActivity.java [hunk 3]
```diff
@@ -89,79 +95,130 @@
...
+        teleopDataStringList = new ArrayList<>();
+
+    }
+
+
+    @Override
+    protected void onResume(){
+        super.onResume();
+
+        teleopDataStringList.clear();
```
Comment (KyletheWolff) [inline app/src/main/java/com/example/vande/scouting2017/TeleopActivity.java]: Why do we need to clear list onResume()

File: app/src/main/java/com/example/vande/scouting2017/TeleopActivity.java [hunk 4]
```diff
@@ -89,79 +95,130 @@
...
+                }
+            }
+        }
+        return false;
     }
 
     public void saveData(View view) throws IOException {
-        final Context context = this;
         String state;
         state = Environment.getExternalStorageState();
```
Comment (KyletheWolff) [inline app/src/main/java/com/example/vande/scouting2017/TeleopActivity.java]: just set String state = Environment.getExternalStorageState(); on one line

File: app/src/main/java/com/example/vande/scouting2017/TeleopActivity.java [hunk 5]
```diff
@@ -89,79 +95,130 @@
...
+        }else if (StringUtils.isEmptyOrNull(getTextInputLayoutString(teleopLowFuelInputLayout))) {
+            teleopLowFuelInputLayout.setError("Enter Low Fuel Scored");
+            ViewUtils.requestFocus(teleopLowFuelInputLayout, this);
+        }else if (StringUtils.isEmptyOrNull(getTextInputLayoutString(climbTimeInputLayout))) {
+            climbTimeInputLayout.setError("Enter Climb Time");
+            ViewUtils.requestFocus(climbTimeInputLayout, this);
+        }else {
+            allInputsPassed = true;
+        }
+
```
Comment (KyletheWolff) [inline app/src/main/java/com/example/vande/scouting2017/TeleopActivity.java]: You need this line under the else
```
 if (!allInputsPassed) {
            return;
        }
```

**Comment by KyletheWolff** [comment]
Update the activity_telop as well to make it look the same as the other

File: app/src/main/java/com/example/vande/scouting2017/TeleopActivity.java [hunk 6]
```diff
@@ -226,8 +287,7 @@ public void saveData(View view) throws IOException {
             Toast.makeText(getApplicationContext(),"message Saved", Toast.LENGTH_LONG).show();
 
 
-
-                Intent intent = new Intent(context, AutonActivity.class);
+                Intent intent = new Intent(this, AutonActivity.class);
```
Comment (KyletheWolff) [inline app/src/main/java/com/example/vande/scouting2017/TeleopActivity.java:289]: Remove the intent and startActivity and below the clearData call put finish();

Starting activity again is going to keep creating intents and a backlog of activites so your back button will not function right.

File: app/src/main/java/com/example/vande/scouting2017/TeleopActivity.java (hunk 5 above)
Comment (vanderpm) [reply app/src/main/java/com/example/vande/scouting2017/TeleopActivity.java]: I added this after the pr

Comment (KyletheWolff) [reply app/src/main/java/com/example/vande/scouting2017/TeleopActivity.java]: Can you push it up

File: app/src/main/java/com/example/vande/scouting2017/TeleopActivity.java (hunk 6 above)
Comment (KyletheWolff) [reply app/src/main/java/com/example/vande/scouting2017/TeleopActivity.java:289]: This needs removed

**Pull request closed without merging**